cobbler/actions/buildiso/__init__.py

```diff
@@ -0,0 +1,221 @@
+
+        :param api: The API instance which holds all information about objects in Cobbler.
+        """
+        self.api = api
+        self.distmap = {}
+        self.distctr = 0
+        self.logger = logging.getLogger()
+
+        self.profiles: List[str] = []
+        self.isolinuxdir = ""
+
+        # grab the header from buildiso.header file
+        with open(os.path.join(self.api.settings().iso_template_dir, "buildiso.template")) as header_src:
+            self.iso_template = header_src.read()
+
+    def copy_boot_files(self, distro, destdir: str, prefix: str = ""):
+        """Copy kernel/initrd to destdir with (optional) newfile prefix
+
+        :param distro: Distro object to return the boot files for.
+        :param destdir: The destionation direcotry.
+        :param prefix: The file prefix.
+        """
+        self.logger.info("copying kernels and initrds for standalone distro")
+        if not os.path.exists(distro.kernel):
+            raise FileNotFoundError("path does not exist: %s" % distro.kernel)
+        if not os.path.exists(distro.initrd):
+            raise FileNotFoundError("Path does not exist: %s" % distro.initrd)
+        if not prefix:
+            shutil.copyfile(distro.kernel, os.path.join(destdir, "%s" % os.path.basename(distro.kernel)))
+            shutil.copyfile(distro.initrd, os.path.join(destdir, "%s" % os.path.basename(distro.initrd)))
+        else:
+            shutil.copyfile(distro.kernel, os.path.join(destdir, "%s.krn" % prefix))
+            shutil.copyfile(distro.initrd, os.path.join(destdir, "%s.img" % prefix))
+
+    def filter_profiles(self, selected_items: List[str] = None) -> list:
+        """
+        Return a list of valid profile objects selected from all profiles by name, or everything if ``selected_items``
+        is empty.
+
+        :param selected_items: A list of names to include in the returned list.
+        :return: A list of valid profiles. If an error occurred this is logged and an empty list is returned.
+        """
+        if selected_items is None:
+            selected_items = []
+        return self.filter_items(list(self.api.profiles()), selected_items)
+
+    def filter_items(self, all_objs: list, selected_items: List[str]) -> list:
+        """Return a list of valid profile or system objects selected from all profiles or systems by name, or everything
+        if selected_items is empty.
+
+        :param all_objs: A list of items to filter.
+        :param selected_items: The list of names
+        :raises ValueError: Second option that this error is raised
+                            when the list of filtered systems or profiles is empty.
+        :return: A list of valid profiles OR systems. If an error occurred this is logged and an empty list is returned.
+        """
+        # TODO: Make use of the find function to have less code.
+        all_objs.sort(key=lambda item: item.name)
```

Comment:
This changes (sorts) the passed in list, is that okay for the caller? If not, the loop below could be written as `for obj in sorted(all_objs, key= lamba item: item.name)`

Reply:
Yes. Because in the end we are using the lists to generate a Boot-Menu this is desired in all cases.